p24/src/bad_lifetimes.rs

```diff
@@ -0,0 +1,53 @@
+fn nth_item<'a>(data: &'a [usize], n: &usize) -> &'a usize {
+    &data[*n]
+}
+
+fn increased_by_first_item<'b>(data: &[usize], n: &'b mut usize) -> &'b mut usize {
+    *n += data[0];
+    n
+}
+
+struct TwoValues<'a> {
```

Comment:
Note that you can use two different lifetimes in a struct. But this works as well for the provided code.

Reply:
Applied your feedback, thanks